beacon-chain/sync/data_columns.go

```diff
@@ -0,0 +1,253 @@
+package sync
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/libp2p/go-libp2p/core"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/peerdas"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/db/filesystem"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/p2p"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/p2p/types"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/startup"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/sync/verify"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/verification"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/interfaces"
+	eth "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/runtime/logging"
+	"github.com/prysmaticlabs/prysm/v5/runtime/version"
+)
+
+// RequestDataColumnSidecars sends a data column sidecars by root request to one
+// or more peers that can provide the needed data columns.
+func RequestDataColumnSidecars(
```

Comment:
I think there is a big issue with the current implementation.
If all requested data columns are not effectively retrieved, this function returns without any error.
It's quite unfortunate, since the caller functions do not check that.

==> This function should implement a retries mechanism to fetch all columns if the first selected set of peers is not able to deliver it.
We can imagine a system like:
- First, we do as today. If all requested columns are retrieved ==> OK
- If not, remove the faulty peers for our list, and try with other peers.

If all possibilities are exhausted, then return on error.